cmds/dhclient/dhcp6client/packetSock.go

```diff
@@ -0,0 +1,159 @@
+package dhcp6client
+
+import (
+	"bytes"
+	"encoding/binary"
+	"errors"
+	"fmt"
+
+	"github.com/mdlayher/dhcp6"
+	"golang.org/x/sys/unix"
+)
+
+type packetSock struct {
+	fd      int
+	ifindex int
+}
+
+var bcastMAC = []byte{255, 255, 255, 255, 255, 255}
+
+func NewPacketSock(ifindex int) (*packetSock, error) {
+	fd, err := unix.Socket(unix.AF_PACKET, unix.SOCK_DGRAM, int(swap16(unix.ETH_P_IPV6)))
+	if err != nil {
+		return nil, err
+	}
+	addr := unix.SockaddrLinklayer{
+		Ifindex:  ifindex,
+		Protocol: swap16(unix.ETH_P_IPV6),
+	}
+	if err = unix.Bind(fd, &addr); err != nil {
+		return nil, err
+	}
+	return &packetSock{
+		fd:      fd,
+		ifindex: ifindex,
+	}, nil
+}
+
+// Write dhcpv6 requests
+func (pc *packetSock) Write(pb []byte) error {
+	// Define linke layer
+	lladdr := unix.SockaddrLinklayer{
+		Ifindex:  pc.ifindex,
+		Protocol: swap16(unix.ETH_P_IPV6),
+		Halen:    uint8(len(bcastMAC)),
+	}
+	copy(lladdr.Addr[:], bcastMAC)
+
+	// Wrap up request
+	//req, err := dhcp6.ParseRequest(pb, addr)
+	//if err != nil {
+	//	return err
+	//}
+
+	//rb, err := MarshalBinary(req)
+	//if err != nil {
+	//	return err
+	//}
+	// Send out request from link layer
+	return unix.Sendto(pc.fd, pb, 0, &lladdr)
+}
+
+func (pc *packetSock) ReadFrom() {
+	fmt.Printf("starts reading\n")
+	pb := make([]byte, 200) // pkt of size 100 bytes, for now
+	n, _, err := unix.Recvfrom(pc.fd, pb, 0)
+	packet := dhcp6.Packet{}
+	UnmarshalBinary(&packet, pb)
+	fmt.Printf("response: %v\n", packet)
+	fmt.Printf("read from server: %v, %v, %v\n", n, pb, err)
+}
+
+func (pc *packetSock) Close() error {
+	return unix.Close(pc.fd)
+}
+
+func UnmarshalBinary(p *dhcp6.Packet, b []byte) error {
```

Comment:
dhcp6.Packet already has an UnmarshalBinary function. Why redefine it here? In fact, this looks to be copied from there? https://github.com/mdlayher/dhcp6/blob/master/packet.go#L43

It seems like we don't need this or parseOptions at all?